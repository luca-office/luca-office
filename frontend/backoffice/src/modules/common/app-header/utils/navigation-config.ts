import {LucaTFunction} from "shared/translations"
import {AppMode} from "../../../../enums"
import {Route} from "../../../../routes"

export interface HeaderMenuItem {
  readonly label: string
  readonly route: Route
  readonly disabled?: boolean
  readonly showSeparator?: boolean
}

export const getHeaderNavigationItems = (
  appMode: AppMode,
  t: LucaTFunction,
  userMayAdministrateRScripts: boolean
): HeaderMenuItem[] => {
  switch (appMode) {
    case AppMode.MANAGER: {
      return []
    }
    case AppMode.EDITOR:
      return [
        {label: t("navigation__scenarios"), route: Route.Scenarios},
        {label: t("navigation__questionnaires"), route: Route.Questionnaires, showSeparator: true},
        {label: t("navigation__reference_books"), route: Route.ReferenceBookChapters},
        {label: t("navigation__sample_companies"), route: Route.SampleCompanies},
        {label: t("navigation__events"), route: Route.Events},
        ...(userMayAdministrateRScripts ? [{label: t("navigation__r_scripts"), route: Route.RScripts}] : [])
      ]
    default:
      return []
  }
}
