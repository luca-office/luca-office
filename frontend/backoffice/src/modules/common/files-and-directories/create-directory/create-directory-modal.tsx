import {css} from "@emotion/react"
import * as React from "react"
import {UseFormMethods} from "react-hook-form"
import {Modal, Overlay, TextInput} from "shared/components"
import {InputType} from "shared/enums"
import {spacingMedium, spacingSmall} from "shared/styles"
import {LucaI18nLangKey, useLucaTranslation} from "shared/translations"

export interface DirectoryCreationForm {
  readonly name: string
}

export interface CreateDirectoryModalProps {
  readonly onDismiss: () => void
  readonly titleKey: LucaI18nLangKey
  readonly formMethods: UseFormMethods<DirectoryCreationForm>
  readonly onSubmit: (formValues: DirectoryCreationForm) => void
  readonly createDirectoryLoading: boolean
}

export const CreateDirectoryModal: React.FC<CreateDirectoryModalProps> = ({
  onDismiss,
  titleKey,
  onSubmit,
  formMethods,
  createDirectoryLoading
}) => {
  const {t} = useLucaTranslation()
  const {register, handleSubmit, errors} = formMethods

  return (
    <Overlay>
      <Modal
        title={t(titleKey)}
        onDismiss={onDismiss}
        customStyles={styles.modal}
        onConfirm={handleSubmit(onSubmit)}
        confirmButtonDisabled={createDirectoryLoading}>
        <div css={styles.inputWrapper}>
          <TextInput
            ref={register({
              required: true
            })}
            placeholderKey="files_and_directories__create_main_directory_modal_name_placeholder"
            customStyles={styles.input}
            name="name"
            hasValidationError={!!errors?.name}
            type={InputType.text}
            labelKey={"files_and_directories__create_main_directory_modal_name_title"}
            autoFocus={true}
          />
        </div>
      </Modal>
    </Overlay>
  )
}

const styles = {
  modal: css({width: 600}),
  inputWrapper: css({
    marginTop: spacingMedium
  }),
  input: css({
    width: "100%",
    marginBottom: spacingSmall
  })
}
