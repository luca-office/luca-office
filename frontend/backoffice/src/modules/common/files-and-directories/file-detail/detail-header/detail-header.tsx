import {css} from "@emotion/react"
import * as React from "react"
import {CardHeader, DeleteOrArchiveEntityButton, Heading, Icon} from "shared/components"
import {HeadingLevel, IconName} from "shared/enums"
import {DeleteEntityHook} from "shared/models"
import {Flex, spacing, spacingMedium, spacingSmall, textEllipsis} from "shared/styles"

export interface FileDetailHeaderProps {
  readonly icon: IconName
  readonly title: string
  readonly entityId: UUID
  readonly deleteHook?: (id: string) => DeleteEntityHook
  readonly onDelete?: () => void
  readonly disabled: boolean
}

export const DetailHeader: React.FC<FileDetailHeaderProps> = ({
  icon,
  title,
  entityId,
  deleteHook,
  onDelete,
  disabled
}) => {
  return (
    <CardHeader customStyles={styles.header} hasShadow hasGreyBackground>
      <div css={styles.title}>
        <Icon customStyles={styles.icon} name={icon} />
        <Heading customStyles={textEllipsis} level={HeadingLevel.h3}>
          {title}
        </Heading>
      </div>
      <div css={[Flex.row, {flexShrink: 0}]}>
        {deleteHook && (
          <DeleteOrArchiveEntityButton
            disabled={disabled}
            entityId={entityId}
            useDeleteHook={deleteHook}
            onSuccess={onDelete}
          />
        )}
      </div>
    </CardHeader>
  )
}

const styles = {
  header: css({
    display: "flex",
    justifyContent: "space-between"
  }),
  title: css({
    display: "flex",
    alignItems: "center",
    minWidth: 0
  }),
  viewerButton: css({
    marginRight: spacingSmall,
    flexShrink: 0,
    width: "auto",
    padding: spacing(0, spacingMedium)
  }),
  icon: css({
    marginRight: spacingSmall
  })
}
