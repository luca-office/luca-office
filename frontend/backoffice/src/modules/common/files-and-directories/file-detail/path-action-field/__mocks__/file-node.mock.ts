import {FileNode} from "shared/components"
import {BinaryType, NodeType} from "shared/enums"

export const fileNodeMock: FileNode = {
  binaryFileId: "spfosdkfsd",
  binaryFileUrl: "url@url.de",
  binaryType: BinaryType.Image,
  id: "sfpoksdf-fposjfd",
  name: "test-file",
  parentId: "sfopksdf-fsdpfosdk",
  tags: [],
  type: NodeType.File
}
