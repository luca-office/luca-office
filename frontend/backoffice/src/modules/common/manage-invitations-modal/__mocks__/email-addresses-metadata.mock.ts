import {EmailAddressesMetadata} from "../models/email-addresses-metadata"

export const emailAddressesMetadataMock: EmailAddressesMetadata = {
  validAddresses: [
    "testmail@test.de",
    "cap@cap3.de",
    "testmail@test.de",
    "testmai2l@test.de",
    "cap3@cap3.de",
    "cap@cap3.de"
  ],
  uniqueAddresses: ["testmail@test.de", "cap@cap3.de", "testmai2l@test.de", "cap3@cap3.de"],
  hasInvalidAddress: false,
  duplicatedAddresses: ["testmail@test.de"],
  alreadyInvitedAddresses: ["cap@cap3.de"]
}
