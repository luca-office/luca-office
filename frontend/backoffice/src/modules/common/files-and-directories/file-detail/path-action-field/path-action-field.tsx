import {css} from "@emotion/react"
import * as React from "react"
import {Icon, Text, Tooltip} from "shared/components"
import {IconName} from "shared/enums"
import {DirectoryNode} from "shared/models"
import {Flex, fontColorLight, FontWeight, primaryColor, spacingTiny, textEllipsis, TextSize} from "shared/styles"
import {WithLucaTranslation} from "shared/translations"
import {Option} from "shared/utils"

export interface PathActionFieldProps extends WithLucaTranslation {
  readonly parentDirectory: Option<DirectoryNode>
  readonly disabled: boolean
  readonly onMoveClick: () => void
  readonly directoryLabel?: string
  readonly tooltipLabel?: string
}

export const PathActionField: React.FC<PathActionFieldProps> = ({
  parentDirectory,
  onMoveClick,
  disabled,
  t,
  directoryLabel = t("files_and_directories__move_directory_main_directory"),
  tooltipLabel = t("files_and_directories__disabled_tooltip")
}) => (
  <Tooltip title={tooltipLabel} inactive={!disabled}>
    <div css={styles.directoryPathWrapper}>
      <div css={[Flex.row, {minWidth: 0}]}>
        <Icon customStyles={styles.icon} name={parentDirectory.isEmpty() ? IconName.HardDrive : IconName.Folder} />
        <Text customStyles={textEllipsis} size={TextSize.Medium}>
          {parentDirectory.map(directory => directory.name).getOrElse(directoryLabel)}
        </Text>
      </div>
      <Text
        onClick={() => !disabled && onMoveClick()}
        customStyles={styles.directoryPath(disabled)}
        size={TextSize.Medium}>
        {t("files_and_directories__move_directory")}
      </Text>
    </div>
  </Tooltip>
)

const styles = {
  directoryPathWrapper: css([
    Flex.row,
    {
      justifyContent: "space-between"
    }
  ]),
  icon: css({
    marginRight: spacingTiny
  }),
  directoryPath: (isDisabled: boolean) =>
    css({
      color: isDisabled ? fontColorLight : primaryColor,
      fontWeight: FontWeight.Bold,
      cursor: isDisabled ? "not-allowed" : "pointer"
    }),
  contentList: css({
    marginTop: spacingTiny
  })
}
