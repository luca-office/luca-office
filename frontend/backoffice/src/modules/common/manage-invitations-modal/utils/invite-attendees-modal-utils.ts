import {intersection, uniq} from "lodash-es"
import {emailRegexPattern} from "shared/utils"
import {EmailAddressesMetadata} from "../models/email-addresses-metadata"

export const getEmailAddressesMetadata = (
  addressesSeparatedWithSemicolon: string,
  alreadyInvitedAddresses: string[]
): EmailAddressesMetadata => {
  const emailAddresses = addressesSeparatedWithSemicolon.split(";").map(address => address.trim())

  const validAddresses = emailAddresses.filter(
    address => emailRegexPattern.test(address) && !alreadyInvitedAddresses.includes(address)
  )

  const uniqueAddresses = uniq(validAddresses)
  const duplicatedAddresses = validAddresses.reduce(
    (accumulator, email, index) =>
      validAddresses.filter(address => address === email).length > 1 && index < validAddresses.lastIndexOf(email)
        ? [...accumulator, email]
        : accumulator,
    [] as string[]
  )

  return {
    validAddresses,
    uniqueAddresses,
    hasInvalidAddress: emailAddresses.length !== validAddresses.length && validAddresses.length > 0,
    duplicatedAddresses,
    alreadyInvitedAddresses: intersection(alreadyInvitedAddresses, emailAddresses)
  }
}

export const getOnlyNewSurveyInvitations = (creations: {email: string}[], existingEmailAddresses: string[]) =>
  creations.filter(creation => !existingEmailAddresses.includes(creation.email))
