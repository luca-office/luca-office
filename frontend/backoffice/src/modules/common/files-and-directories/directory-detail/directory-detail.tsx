import {CSSInterpolation} from "@emotion/serialize"
import React, {Dispatch, Fragment, SetStateAction} from "react"
import {Card, CardContent, FormFieldLabel, Heading, ReadonlyActionField, Tooltip} from "shared/components"
import {HeadingLevel, IconName, UploadFileType as FileType} from "shared/enums"
import {TextDocumentCreation} from "shared/graphql/generated/globalTypes"
import {DeleteEntityHook, Directory, DirectoryNode, File, UploadBinary} from "shared/models"
import {TextDocumentCreationInfo} from "shared/models/text-document-creation-info"
import {CustomStyle, FontWeight} from "shared/styles"
import {LucaI18nLangKey, useLucaTranslation} from "shared/translations"
import {Option} from "shared/utils"
import {InlineEditableHeaderContainer, UploadFileModal} from "../../../../components"
import {HintText} from "../../../../components/hint-text/hint-text"
import {DirectoryUpdate} from "../../../../graphql/generated/globalTypes"
import {detailViewStyles} from "../../../files-and-directories/detail/detail-styles"
import {DirectoryContentList} from "../directory-content-list/directory-content-list"
import {DetailHeader} from "../file-detail/detail-header/detail-header"
import {PathActionField} from "../file-detail/path-action-field/path-action-field"
import {MoveOverlay} from "../move/move-overlay"

interface DirectoryConfig {
  readonly directory: Directory
  readonly parentDirectory: Option<DirectoryNode>
  readonly subdirectories: Directory[]
  readonly allDirectories: Directory[]
  readonly updateDirectory: (id: UUID, update: DirectoryUpdate) => Promise<Option<Directory>>
  readonly selectDirectory: (directoryId: UUID) => void
  readonly deselectDirectory: () => void
  readonly expandDirectoryId?: (id: UUID) => void
  readonly directoryLabel?: string
  readonly deleteDirectoryHook: () => DeleteEntityHook
  readonly renderCreateDirectoryModal: () => React.ReactNode
}

interface FileConfig {
  readonly files: File[]
  readonly allFiles: File[]
  readonly selectFile: (fileId: UUID) => void
  readonly deleteFileHook: () => DeleteEntityHook
  readonly createFilesFromBinaries: (uploadedBinaries: UploadBinary[], directoryId: UUID) => void
}

interface CreateFileOverlayConfig {
  readonly isCreateFileOverlayVisible: boolean
  readonly showCreateFileOverlay: () => void
  readonly hideCreateFileOverlay: () => void
}

interface CreateSubdirectoryOverlayConfig {
  readonly isCreateSubdirectoryOverlayVisible: boolean
  readonly showCreateSubdirectoryOverlay: () => void
}

interface UpdateDirectoryOverlayConfig {
  readonly isUpdateDirectoryOverlayVisible: boolean
  readonly setIsUpdateDirectoryOverlayVisible: Dispatch<SetStateAction<boolean>>
}

interface DisabledConfig {
  readonly isMoveDisabled?: boolean
  readonly isCreateSubDirectoryDisabled?: boolean
  readonly isContentListEntityDisabled?: (entityId: UUID) => boolean
  readonly isLabelDisabled?: boolean
  readonly isDeleteDisabled?: boolean
}

interface TooltipConfig {
  readonly disabledTooltipLabel?: string
  readonly tooltipTitleKey?: LucaI18nLangKey
}

export interface DirectoryDetailProps extends CustomStyle {
  readonly createFileOverlayConfig: CreateFileOverlayConfig
  readonly createSubdirectoryOverlayConfig: CreateSubdirectoryOverlayConfig
  readonly dataLoading: boolean
  readonly directoryConfig: DirectoryConfig
  readonly disabled: boolean
  readonly fileConfig: FileConfig
  readonly updateDirectoryOverlayConfig: UpdateDirectoryOverlayConfig
  readonly updateInProgress: boolean
  readonly customContentStyles?: CSSInterpolation
  readonly disabledConfig?: DisabledConfig
  readonly navigateToSampleCompany?: () => void
  readonly sampleCompanyId?: UUID
  readonly showRelevance?: boolean
  readonly tooltipConfig?: TooltipConfig
  readonly customDirectoryIcon?: IconName
  readonly createTextDocument?: (textDocumentCreationInfo: TextDocumentCreationInfo) => void
}

export const DirectoryDetail: React.FC<DirectoryDetailProps> = ({
  createFileOverlayConfig,
  createSubdirectoryOverlayConfig,
  customContentStyles,
  customDirectoryIcon,
  customStyles,
  dataLoading,
  directoryConfig,
  disabled,
  disabledConfig,
  fileConfig,
  sampleCompanyId,
  showRelevance,
  tooltipConfig,
  updateDirectoryOverlayConfig,
  updateInProgress,
  navigateToSampleCompany,
  createTextDocument
}) => {
  const {t} = useLucaTranslation()

  const {
    directory,
    parentDirectory,
    subdirectories,
    allDirectories,
    updateDirectory,
    selectDirectory,
    deselectDirectory,
    expandDirectoryId,
    directoryLabel,
    deleteDirectoryHook,
    renderCreateDirectoryModal
  } = directoryConfig
  const {files, allFiles, selectFile, deleteFileHook, createFilesFromBinaries} = fileConfig
  const {isCreateFileOverlayVisible, showCreateFileOverlay, hideCreateFileOverlay} = createFileOverlayConfig
  const {isCreateSubdirectoryOverlayVisible, showCreateSubdirectoryOverlay} = createSubdirectoryOverlayConfig
  const {isUpdateDirectoryOverlayVisible, setIsUpdateDirectoryOverlayVisible} = updateDirectoryOverlayConfig

  const isMoveDisabled = disabledConfig?.isMoveDisabled ?? false
  const isCreateSubDirectoryDisabled = disabledConfig?.isCreateSubDirectoryDisabled ?? false
  const isContentListEntityDisabled = disabledConfig?.isContentListEntityDisabled
  const isLabelDisabled = disabledConfig?.isLabelDisabled ?? false
  const isDeleteDisabled = disabledConfig?.isDeleteDisabled ?? false

  const disabledTooltipLabel = tooltipConfig?.disabledTooltipLabel
  const tooltipTitleKey = tooltipConfig?.tooltipTitleKey ?? "files_and_directories__disabled_tooltip"

  const handleUpdate = (update: Partial<DirectoryUpdate>) => {
    if (updateInProgress) {
      return Promise.reject()
    }
    return updateDirectory(directory.id, {
      name: update.name ? update.name : directory.name,
      parentDirectoryId: update.parentDirectoryId ? update.parentDirectoryId : directory.parentDirectoryId
    })
  }

  const createTextDocumentInDirectory = (textDocument: TextDocumentCreation, title: string) => {
    createTextDocument?.({textDocument, title, directoryId: directory.id})
  }

  return (
    <Fragment>
      <Card customStyles={customStyles}>
        <DetailHeader
          deleteHook={deleteDirectoryHook}
          onDelete={deselectDirectory}
          entityId={directory.id}
          title={directory.name}
          icon={customDirectoryIcon || IconName.Folder}
          disabled={disabled || isDeleteDisabled}
        />

        <CardContent customStyles={[detailViewStyles.content, customContentStyles]}>
          {sampleCompanyId && (
            <HintText
              text={t("files_and_directories__sample_company_directory_hint")}
              customStyles={detailViewStyles.hintText}
            />
          )}
          <div css={detailViewStyles.nameAndDirectoryWrapper}>
            <div>
              <FormFieldLabel textKey="files_and_directories__rename_directory_label" />
              <Tooltip title={disabledTooltipLabel ?? t(tooltipTitleKey)} inactive={!disabled}>
                <InlineEditableHeaderContainer
                  disabled={disabled || isLabelDisabled}
                  onConfirm={name => handleUpdate({name})}
                  customStyles={detailViewStyles.directoryNameInput}
                  text={directory.name}
                />
              </Tooltip>
            </div>
            <div>
              <FormFieldLabel
                textKey={
                  sampleCompanyId
                    ? "files_and_directories__sample_company_label"
                    : "files_and_directories__move_directory_label"
                }
              />
              <ReadonlyActionField
                customStyles={detailViewStyles.actionField}
                renderValue={() =>
                  sampleCompanyId ? (
                    directory.name
                  ) : (
                    <PathActionField
                      onMoveClick={() => setIsUpdateDirectoryOverlayVisible(true)}
                      t={t}
                      disabled={disabled || isMoveDisabled}
                      tooltipLabel={disabledTooltipLabel ?? t(tooltipTitleKey)}
                      parentDirectory={parentDirectory}
                      directoryLabel={directoryLabel}
                    />
                  )
                }
                disabled={!sampleCompanyId && (disabled || isMoveDisabled)}
                onClick={sampleCompanyId ? navigateToSampleCompany : undefined}
                buttonLabel={sampleCompanyId ? t("files_and_directories__sample_company_button") : undefined}
              />
            </div>
          </div>

          <Heading
            level={HeadingLevel.h3}
            fontWeight={FontWeight.Bold}
            customStyles={detailViewStyles.contentListHeading}>
            {t("files_and_directories__directory_content_list_heading")} ({subdirectories.length + files.length})
          </Heading>
          <DirectoryContentList
            customStyles={detailViewStyles.contentList}
            directories={subdirectories}
            files={files}
            onClickDirectory={selectDirectory}
            onClickFile={selectFile}
            onCreateFile={showCreateFileOverlay}
            onCreateSubdirectory={showCreateSubdirectoryOverlay}
            disabled={disabled}
            tooltipLabel={t(tooltipTitleKey)}
            deleteDirectoryHook={deleteDirectoryHook}
            deleteFileHook={deleteFileHook}
            isCreateSubDirectoryDisabled={isCreateSubDirectoryDisabled}
            isTableEntityDisabled={isContentListEntityDisabled}
            showRelevance={showRelevance}
          />
        </CardContent>
      </Card>

      {isCreateFileOverlayVisible ? (
        <UploadFileModal
          acceptedFileTypes={[FileType.Graphic, FileType.Video, FileType.PDF, FileType.TableCalculation]}
          onBinariesSuccessfullyUploaded={binaries => {
            hideCreateFileOverlay()
            createFilesFromBinaries(binaries, directory.id)
          }}
          onDismiss={hideCreateFileOverlay}
          createTextDocument={createTextDocumentInDirectory}
        />
      ) : null}

      {isCreateSubdirectoryOverlayVisible && renderCreateDirectoryModal()}

      {isUpdateDirectoryOverlayVisible && (
        <MoveOverlay
          onDismiss={() => setIsUpdateDirectoryOverlayVisible(false)}
          file={Option.none()}
          parentDirectory={parentDirectory}
          name={directory.name}
          directory={Option.of(directory)}
          onFilesSuccessfullyMoved={targetDirectoryIds => {
            if (expandDirectoryId) {
              targetDirectoryIds.forEach(directoryId => expandDirectoryId(directoryId))
            }
            setIsUpdateDirectoryOverlayVisible(false)
          }}
          directories={allDirectories}
          files={allFiles}
          dataLoading={dataLoading}
        />
      )}
    </Fragment>
  )
}
