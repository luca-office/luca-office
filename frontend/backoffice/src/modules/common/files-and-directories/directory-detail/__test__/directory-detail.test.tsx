// importing from direct file because of issues of babel loader and spyOn
import {MockedProvider} from "@apollo/client/testing"
import {mount} from "enzyme"
import * as React from "react"
import {create} from "react-test-renderer"
import {Card, CardContent, FormFieldLabel} from "shared/components"
import {filesMock, sampleCompaniesMock} from "shared/graphql/__mocks__"
import {Option} from "shared/utils"
import {InlineEditableHeaderContainer} from "../../../../../components"
import {HintText} from "../../../../../components/hint-text/hint-text"
import {directoriesForScenarioMock} from "../../../../../graphql/__mocks__"
import {CreateDirectoryModal} from "../../create-directory/create-directory-modal"
import {DetailHeader} from "../../file-detail/detail-header/detail-header"
import {PathActionField} from "../../file-detail/path-action-field/path-action-field"
import {DirectoryDetail, DirectoryDetailProps} from "../directory-detail"

const directoryMock = directoriesForScenarioMock[0]

const deleteEntityHookMock = jest.fn(() => ({
  deleteEntity: jest.fn(() => Promise.resolve()),
  deleteEntityLoading: false
}))

const getComponent = (props?: Partial<DirectoryDetailProps>) => (
  <MockedProvider>
    <DirectoryDetail
      disabled={false}
      navigateToSampleCompany={jest.fn}
      directoryConfig={{
        directory: directoryMock,
        parentDirectory: Option.none(),
        subdirectories: [],
        allDirectories: directoriesForScenarioMock,
        updateDirectory: jest.fn(() => Promise.resolve(Option.of(directoryMock))),
        selectDirectory: jest.fn(),
        deselectDirectory: jest.fn(),
        deleteDirectoryHook: deleteEntityHookMock,
        renderCreateDirectoryModal: jest.fn()
      }}
      fileConfig={{
        files: [],
        allFiles: filesMock,
        selectFile: jest.fn(),
        deleteFileHook: deleteEntityHookMock,
        createFilesFromBinaries: jest.fn()
      }}
      createFileOverlayConfig={{
        isCreateFileOverlayVisible: false,
        showCreateFileOverlay: jest.fn(),
        hideCreateFileOverlay: jest.fn()
      }}
      createSubdirectoryOverlayConfig={{
        isCreateSubdirectoryOverlayVisible: false,
        showCreateSubdirectoryOverlay: jest.fn()
      }}
      updateDirectoryOverlayConfig={{
        isUpdateDirectoryOverlayVisible: false,
        setIsUpdateDirectoryOverlayVisible: jest.fn()
      }}
      updateInProgress={false}
      dataLoading={false}
      {...props}
    />
  </MockedProvider>
)

describe("Directory Detail View", () => {
  it("renders correctly", () => {
    const component = create(getComponent())
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("has correct structure", () => {
    const component = mount(getComponent())
    expect(component.find(Card)).toHaveLength(1)
    expect(component.find(CardContent)).toHaveLength(1)
    expect(component.find(DetailHeader)).toHaveLength(1)
    expect(component.find(FormFieldLabel)).toHaveLength(2)
    expect(component.find(CreateDirectoryModal)).toHaveLength(0)
    expect(component.find(InlineEditableHeaderContainer)).toHaveLength(1)
    expect(component.find(PathActionField)).toHaveLength(1)
  })
  it("has correct structure for sample company", () => {
    const component = mount(getComponent({sampleCompanyId: sampleCompaniesMock[0].id}))
    expect(component.find(Card)).toHaveLength(1)
    expect(component.find(CardContent)).toHaveLength(1)
    expect(component.find(DetailHeader)).toHaveLength(1)
    expect(component.find(FormFieldLabel)).toHaveLength(2)
    expect(component.find(CreateDirectoryModal)).toHaveLength(0)
    expect(component.find(InlineEditableHeaderContainer)).toHaveLength(1)
    expect(component.find(PathActionField)).toHaveLength(0)
  })

  it("triggers update name correctly", () => {
    const updateDirectory = jest.fn()
    const component = mount(
      getComponent({
        directoryConfig: {
          directory: directoryMock,
          parentDirectory: Option.none(),
          subdirectories: [],
          allDirectories: directoriesForScenarioMock,
          updateDirectory,
          selectDirectory: jest.fn(),
          deselectDirectory: jest.fn(),
          deleteDirectoryHook: deleteEntityHookMock,
          renderCreateDirectoryModal: jest.fn()
        }
      })
    )
    component.find(InlineEditableHeaderContainer).props().onConfirm("test-directory")
    expect(updateDirectory).toHaveBeenCalledTimes(1)
    expect(updateDirectory).toHaveBeenCalledWith(directoryMock.id, {name: "test-directory", parentDirectoryId: null})
  })
})
