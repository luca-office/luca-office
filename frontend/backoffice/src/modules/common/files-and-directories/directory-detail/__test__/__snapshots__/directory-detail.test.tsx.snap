// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Directory Detail View renders correctly 1`] = `
<div
  className="css-63bajj-Card"
>
  <div
    className="card-header css-13nbzsf-CardHeader"
  >
    <div
      className="css-17beenm-DetailHeader"
    >
      <div
        className="css-1lrzij5-Icon"
        dangerouslySetInnerHTML={
          {
            "__html": {},
          }
        }
      />
      <div
        className="css-1i6jmor-Heading"
      >
        Bilder
      </div>
    </div>
    <div
      className="css-ntx7po-DetailHeader"
    >
      <div
        className="css-0"
      >
        <button
          className="css-hnekdl-Button"
          disabled={false}
          onClick={[Function]}
        >
          <div
            className="css-95j2hm-Button"
          >
            <div
              className="css-mj8vo1-Icon"
              dangerouslySetInnerHTML={
                {
                  "__html": {},
                }
              }
            />
          </div>
        </button>
      </div>
    </div>
  </div>
  <div
    className="card-content css-spllt3-CardContent"
  >
    <div
      className="css-g7oywz-DirectoryDetail"
    >
      <div>
        <div
          className="css-13s2sfk-Heading"
        >
          files_and_directories__rename_directory_label
        </div>
        <div
          className="css-ryq9y7-InlineEditableHeader"
        >
          <div>
            <form
              className="css-1c5ej61-InlineEditableHeader"
              onBlur={[Function]}
              onSubmit={[Function]}
            >
              <div
                className="css-pl4s5v-InlineEditableHeader"
                onClick={[Function]}
              >
                <div
                  className="css-1iwo50k"
                  onMouseEnter={[Function]}
                  onMouseLeave={[Function]}
                >
                  <div
                    className="css-nrul0i"
                  >
                    <input
                      className="css-q6vozw"
                      defaultValue="Bilder"
                      disabled={false}
                      onBlur={[Function]}
                      onChange={[Function]}
                      onFocus={[Function]}
                      readOnly={true}
                      type="text"
                    />
                  </div>
                </div>
                <div
                  className="hover-icon css-16vb3lw-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
              </div>
            </form>
          </div>
        </div>
      </div>
      <div>
        <div
          className="css-13s2sfk-Heading"
        >
          files_and_directories__move_directory_label
        </div>
        <div
          className="css-1kz9nmi-ReadonlyActionField"
          data-testid="readonly-action-field"
        >
          <div
            className="css-1bkess4-ReadonlyActionField"
          />
          <div
            className="content-wrapper css-b45n67-ReadonlyActionField"
          >
            <div
              className="content css-1147qp7-ReadonlyActionField"
            >
              <div
                className="css-1czgmen-PathActionField"
              >
                <div
                  className="css-1h826ga-PathActionField"
                >
                  <div
                    className="css-1yjyjy3-Icon"
                    dangerouslySetInnerHTML={
                      {
                        "__html": {},
                      }
                    }
                  />
                  <div
                    className="css-1sbf56p-Text"
                  >
                    files_and_directories__move_directory_main_directory
                  </div>
                </div>
                <div
                  className="css-16ebz09-Text"
                  onClick={[Function]}
                >
                  files_and_directories__move_directory
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      className="css-n9iuhz-Heading"
    >
      files_and_directories__directory_content_list_heading
       (
      0
      )
    </div>
    <div
      className="css-1t69vky-Table"
    >
      <div
        className="css-v9uh6m-Table"
      >
        <div
          className="css-1hhq1hd-Table"
        >
          <div
            className="css-8050ac-Heading"
          >
            files_and_directories__directory_detail__column_header_title
          </div>
        </div>
        <div
          className="css-1v7x2py-Table"
        >
          <div
            className="css-8050ac-Heading"
          >
            files_and_directories__directory_detail__column_header_relevance
          </div>
        </div>
        <div
          className="css-1nqo6qf-Table"
        >
          <div
            className="css-0"
            id="tooltip_42"
          >
            <button
              className="css-pzbp9b-Button"
              disabled={false}
              onClick={[Function]}
            >
              <div
                className="css-95j2hm-Button"
              >
                <div
                  className="css-kuxry5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                files_and_directories__directory_detail_sub_directory
              </div>
            </button>
          </div>
          <div
            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
            role="tooltip"
            style={
              {
                "backgroundColor": "#2e3032",
                "zIndex": 1,
              }
            }
          >
            files_and_directories__create_subdirectory_tooltip
            <div
              className="react-tooltip-arrow styles-module_arrow__K0L3T"
              style={{}}
            />
          </div>
        </div>
        <div
          className="css-111d83p-Table"
        >
          <div
            className="css-vhtthd-Tooltip"
            id="tooltip_42"
          >
            <button
              className="css-pzbp9b-Button"
              disabled={false}
              onClick={[Function]}
            >
              <div
                className="css-95j2hm-Button"
              >
                <div
                  className="css-kuxry5-Icon"
                  dangerouslySetInnerHTML={
                    {
                      "__html": {},
                    }
                  }
                />
                files_and_directories__files
              </div>
            </button>
          </div>
          <div
            className="react-tooltip styles-module_tooltip__mnnfp styles-module_dark__xNqje css-10zyuev-Tooltip"
            role="tooltip"
            style={
              {
                "backgroundColor": "#2e3032",
                "zIndex": 1,
              }
            }
          >
            files_and_directories__create_file_tooltip
            <div
              className="react-tooltip-arrow styles-module_arrow__K0L3T"
              style={{}}
            />
          </div>
        </div>
      </div>
      <div
        className="css-tqgzls-Table"
      >
        <div
          className="table-placeholder css-1v6q0ml-Table"
        >
          files_and_directories__directory_content_list_placeholder
        </div>
      </div>
    </div>
  </div>
</div>
`;
